src/sinks/prometheus/exporter.rs

```diff
@@ -333,8 +338,37 @@ impl MetricNormalize for PrometheusExporterMetricNormalizer {
     }
 }
 
+fn check_auth(req: &Request<Body>, auth: &Option<Auth>) -> bool {
+    if let Some(auth) = auth {
+        let headers = req.headers();
+        if let Some(auth_header) = headers.get(hyper::header::AUTHORIZATION) {
+            let encoded_credentials = match auth {
+                Auth::Basic { user, password } => HeaderValue::from_str(
+                    format!("Basic {}", base64::encode(format!("{}:{}", user, password))).as_str(),
+                ),
+                Auth::Bearer { token } => {
+                    HeaderValue::from_str(format!("Bearer {}", token).as_str())
+                }
+            };
+
+            if let Ok(encoded_credentials) = encoded_credentials {
+                if auth_header != encoded_credentials {
+                    return false;
```

Comment:
Small thing, but you could probably invert the conditional here, `return true` and then just have one `false` at the bottom as the default return value.

Reply:
In this case the check `if let Some(auth) = auth { ...` will not work properly - it will deny all requests, if `auth` is not specified in the config file.

Reply:
I have reorganized if conditions a little bit. Maybe this way will be a little bit better.